### PR TITLE
#22817 intial changes

### LINKED ISSUE #22817: Track event Server side
We are going to user this plugins to send the event from DotCMS server to jitsu:

https://github.com/dotCMS/com.dotcms.jitsu

do we need to include this code into the dotCMS core when we get out with Experiment?

We need to save some more data in each Event like:

- Variant that the visitor is using
- Experiment name.

https://github.com/dotCMS/com.dotcms.jitsu/pull/1/files#diff-57dcb40725e634f6dbc4b3f0eb30c3b305a422567d67d5154707b168e4a98027R135

**Acceptance Criteria:**

- Move the plugin code into core
- Make postman test if it is possible and/or integration test (don't forget the JOB https://github.com/dotCMS/com.dotcms.jitsu/blob/main/src/main/java/com/dotcms/jitsu/webinterceptors/EventLogRunnable.java#L34)
- Include the code to storage the new data

### PR DESCRIPTION
**Pull request by dsilvam**
* Brings plugin to send Jitsu events into core.
* Postman tests against cube.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/jitsu/EventLogRunnable.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import com.dotmarketing.util.Config;
+import java.util.Map;
+import com.dotcms.http.CircuitBreakerUrl;
+import com.dotcms.http.CircuitBreakerUrl.Method;
+import com.dotcms.http.CircuitBreakerUrlBuilder;
+import com.dotmarketing.util.Logger;
+import com.google.common.collect.ImmutableMap;
+
+public class EventLogRunnable implements Runnable {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/jitsu/EventLogRunnable.java:11]: doc

File: dotCMS/src/main/java/com/dotcms/jitsu/EventLogRunnable.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    final String POST_URL = Config.getStringProperty("EVENT_LOG_POSTING_URL",
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/jitsu/EventLogRunnable.java:13]: I don't know if here make sense have a default value, maybe we just should send a error if we don't have any values in EVENT_LOG_POSTING_URL

File: dotCMS/src/main/java/com/dotcms/jitsu/EventLogRunnable.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            "http://localhost:8081/api/v1/event");
+
+    final Map<String, String> POSTING_HEADERS = ImmutableMap.of("content-type", "application/json");
+
+    final String token = Config.getStringProperty("EVENT_LOG_TOKEN",
+            "js.cluster1.customer1.3c7ftd589dwax9eqpz");
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/jitsu/EventLogRunnable.java]: same here

File: dotCMS/src/main/java/com/dotcms/jitsu/EventLogSubmitter.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package com.dotcms.jitsu;
+
+import java.util.concurrent.ThreadPoolExecutor;
+import com.dotcms.concurrent.DotConcurrentFactory;
+import com.dotcms.concurrent.DotConcurrentFactory.SubmitterConfig;
+import com.dotcms.concurrent.DotSubmitter;
+import com.dotmarketing.util.Config;
+
+public class EventLogSubmitter {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/jitsu/EventLogSubmitter.java:9]: doc

File: dotCMS/src/main/java/com/dotcms/jitsu/EventLogWebInterceptor.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class EventLogWebInterceptor implements WebInterceptor {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/jitsu/EventLogWebInterceptor.java:22]: doc

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/jitsu/EventLogRunnable.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/jitsu/EventLogRunnable.java:11]: Done

File: dotCMS/src/main/java/com/dotcms/jitsu/EventLogRunnable.java (hunk 3 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/jitsu/EventLogRunnable.java]: Done

File: dotCMS/src/main/java/com/dotcms/jitsu/EventLogWebInterceptor.java (hunk 5 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/jitsu/EventLogWebInterceptor.java:22]: Done

File: dotCMS/src/main/java/com/dotcms/jitsu/EventLogRunnable.java (hunk 2 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/jitsu/EventLogRunnable.java:18]: This is ok for now. We can change later

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/jitsu/EventLogWebInterceptor.java [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        }
+    }
+
+    public void doOptions(final HttpServletRequest request, final HttpServletResponse response) {
+        response.setHeader("content-length", "0");
+        response.setStatus(200);
+    }
+
+    public void setHeaders(final HttpServletResponse response) {
+        response.addHeader("content-type", "application/javascript; charset=utf-8");
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/jitsu/EventLogWebInterceptor.java:111]: Can we use constants from `javax.ws.rs.core.HttpHeaders`

Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/jitsu/EventLogWebInterceptor.java:111]: Only two of these were available

**Review by victoralfaro-dotcms — APPROVED** [review]

**Pull request merged**